<!--                                         -->
<!--     ロボットベースI/F launch ファイル   -->
<!--     L-MES                               -->
<!--                                         -->
<launch>
  <node pkg="trobot" type="trobot_node" name="trobot_node">
    <param name="trobotVersion"        value="L-MES" />
    <param name="activeConfig"         value="gemini" />
    <param name="serialPortName"       value="/dev/ttyACM0" />
    <param name="wheelRadius"          value="0.0625" />
    <param name="axleTrack"            value="0.415" />
    <param name="rotEncoderResolution" value="1200000" />
    <param name="wheelGearRatio"       value="0.05333" />
    <param name="coeffRpsToCmd"        value="7.20" />    <!-- wheelRadius * 160/(最大速度=5km/h) * (60.0 * 60.0) / 1000.0 -->
    <param name="velAccelMax"          value="32" />
    <param name="velDecelMax"          value="32" />
    <param name="autoRecovery"         value="AUTO" />
    <param name="FrameRate"            value="16" />
    <param name="publish_tf"           value="false" />
    <param name="odom_frame"           value="odom" />
    <param name="base_frame"           value="base_footprint" />


    <!-- <param name="coeffVxCmd"          value="-1.0" /> -->
    <!-- <param name="coeffVaCmd"          value="-1.0" /> -->
    <param name="coeffVxCmd"          value="1.0" />
    <param name="coeffVaCmd"          value="1.0" />
    <param name="coeffXOdo"           value="1.0" />
    <param name="coeffYOdo"           value="1.0" />
    <param name="coeffHeadingOdo"     value="1.0" />
    <param name="coeffHeadingOdoAdd"  value="0.0" />

    <param name="trace_log"           value="false" />

    <remap from="cmd_vel" to="cmd_vel2" />
  </node>
</launch>

<!--                                                                           -->
<!-- パラメータ説明                                                            -->
<!--                                                                           -->
<!-- trobotVersion         : ロボットバージョン                                -->
<!--    (旧伝送フォーマット："OLDrobot",新伝送フォーマット："SCIBOT-X"）       -->
<!-- activeConfig          : システム名                                        -->
<!--    (ログ出力によりConfiguration SET確認用)                                -->
<!-- serialPortName        : シリアルポートの名称                              -->
<!--   （Linux: /dev/ttyUSB0, etc.,  Windows: COM3, etc.）                     -->
<!-- wheelRadius           : 駆動輪の半径 [m]                                  -->
<!-- axleTrack             : 輪距（左右の駆動輪中心間の距離） [m]              -->
<!-- rotEncoderResolution  : 駆動輪モータ軸のエンコーダの分解能 [pulses/rev]   -->
<!-- wheelGearRatio        : 駆動輪の減速機の減速比 [無単位]                   -->
<!-- coeffRpsToCmd : 角速度 [rad/s] から回転速度指令値への変換係数             -->
<!--     KMPH_TO_CMD : I/F仕様書の速度指令値の最大値/最大値の時の時速          -->
<!--     KMPH_TO_MPS : エンコーダカウント値のサンプリング周期                  -->
<!--         = (60.0 * 60.0) / 1000.0                                          -->
<!--     m_wheelRadius * KMPH_TO_CMD * KMPH_TO_MPS                             -->
<!-- アライメント調整                                                          -->
<!-- coeffVxCmd            : 指令値進行方向速度(vx)にかける係数                -->
<!-- coeffVyCmd            : 指令値進行方向直角左向き速度(vy)にかける係数      -->
<!-- coeffvaCmd            : 指令値車体角速度(va)にかける係数                  -->
<!-- coeffXOdo             : オドメトリX座標増分にかける係数                   -->
<!-- coeffYOdo             : オドメトリY座標増分にかける係数                   -->
<!-- coeffHeadingOdo       : オドメトリ向き増分にかける係数                    -->
<!-- coeffHeadingOdoAdd    : オドメトリ向き増分に足しこむ係数                  -->
<!--                                                                           -->
<!-- 1指令周期毎の加速度制限(前回指令値からの速度差分制限)                     -->
<!-- velAccelMax           : 加速度制限                                        -->
<!-- velDecelMax           : 減速度制限                                        -->
<!--                                                                           -->
<!-- autoRecovery          : 緊急停止モードで要因が復旧した時に                -->
<!--                         自動で復旧処理を行う                              -->
<!--                        （ "MANUAL"(default):手動でモード遷移させる、      -->
<!--                           "AUTO"：自動で運転モードに遷移する ）           -->
<!--                                                                           -->
<!-- trace_log             : トレースログ出力フラグ（true:トレースログ出力）   -->
<!--                                                                           -->
